### PR TITLE
Fix for histogram (addressing #2475)

### PR DESCRIPTION
**Pull request by pc494**
### Description of the change
As described in issue (#2475):
- Numpy's histogram functionality offers three types for bins:
`int or sequence of scalars or str`
- The wrapping in hist_tools wasn't set up to handle arrays (or `int`)
- This PR fixes that issue
- The PR also adds a barebones test to prevent regressions

### Progress of the PR
- [x] Change implemented (can be split into several points),
- [ ] update docstring ~(if appropriate)~,
- [ ] update user guide ~(if appropriate)~,
- [x] add tests,
- [x] ready for review.

### Minimal example of the bug fix or the new feature
```python
>>> import hyperspy.api as hs
>>> import numpy as np
>>> s = hs.signals.Signal1D(np.arange(10))
>>> s.get_histogram(bins=3)
```

### CODE REVIEW

**Comment by tjof2** [comment]
Thanks @pc494! Please can you point it to (and probably base off) the `RELEASE_next_patch` branch instead?

**Comment by tjof2** [comment]
Also while you're at it, please can you apply the same fix to the dask version?
https://github.com/pc494/hyperspy/blob/833793109edb3e7b51288eeeff96f05babb21368/hyperspy/misc/hist_tools.py#L172-L180

**Comment by dnjohnstone** [comment]
Just to say - It would be great to have an approximate timeline for a 1.6.1 release when this is merged as this is currently breaking a couple of pyxem tests, please :)

**Comment by tjof2** [comment]
See #2479 for the unrelated Travis test failures, but there's still a remaining issue with the docs build 

> Just to say - It would be great to have an approximate timeline for a 1.6.1 release when this is merged as this is currently breaking a couple of pyxem tests, please :)

cc. @francisco-dlp @ericpre

File: hyperspy/Release.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 # When running setup.py the ".dev" string will be replaced (if possible)
 # by the output of "git describe" if git is available or the git
 # hash if .git is present.
-version = "1.6.0.dev"
+version = "1.7.0.dev"
```
Comment (dnjohnstone) [inline hyperspy/Release.py:28]: Don't forget you're based on the wrong branch here. I.e. should be "1.6.1.dev"

Comment (pc494) [reply hyperspy/Release.py:28]: Thanks, I had changed it on github but not rebased. Dealt with in 72703cb

Comment (tjof2) [reply hyperspy/Release.py:28]: Pretty certain this file shouldn't be showing up at all in the "changed" list in this PR?

**Review by tjof2 — CHANGES_REQUESTED** [review]
Looks great except `Release.py` still seems to be part of the diff.

**Comment by pc494** [comment]
Still plugging away at getting rid of the stray commit

**Comment by thomasaarholt** [comment]
If you're having some hard-to-detangle git trouble, it might be easier to just create a new branch from `hyperspy/RELEASE_next_patch`, manually add in the changes you've done here and then force push to this branch.

**Comment by pc494** [comment]
> If you're having some hard-to-detangle git trouble, it might be easier to just create a new branch from `hyperspy/RELEASE_next_patch`, manually add in the changes you've done here and then force push to this branch.

Thanks, I went with this option in the end, the new PR is #2482

**Comment by thomasaarholt** [comment]
Nice!
What you did is absolutely fine, but what you could have done is* is take the new branch you created for  #2482  and point it at this branch, with something like `git branch --set-upstream-to origin/fix-for-histogram` and then `git push --force` (force because the git commit history is different).

*after backing up this branch just in case - `git branch fix-for-histogram-backup`

**Comment by pc494** [comment]
Good to know, thanks @tjof2!

**Pull request closed without merging**